e2e/config/config.go

```diff
@@ -71,8 +71,17 @@ type Deployer struct {
 	Type string `json:"type"`
 	// Description is a human-readable description of the deployer.
 	Description string `json:"description"`
+	Recipe      Recipe `json:"recipe,omitempty"`
```

Comment:
This is an optional field, so it should be a pointer.

This will never be empty. ramenctl includes the config in the report:
https://github.com/RamenDR/ramenctl/blob/876be1e60bd9aad7a4eccd88cb03da99972c539c/pkg/test/report.go#L30

so we will always see a zero value receipe in all deployers:

```yaml
deployers:
- name: appset
  type: appset
  recipe:
    recipeType: ""
```

With a pointer we will see the recipe only when it is set.

See how we do the same for the build info (missing in odf-cli):
https://github.com/RamenDR/ramenctl/blob/876be1e60bd9aad7a4eccd88cb03da99972c539c/pkg/report/report.go#L46